### PR TITLE
ci: add check for rules docs

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR restores a script that checks the validity of the docs of our rules. This code was copied from the website repository and stripped of all the logic regarding generating the docs. 

Plus, it adds this check inside the CI and `gen-lint`

## Test Plan

CI should pass

### CODE REVIEW

File: xtask/rules_check/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+
+            _ => {
+                // TODO: Implement remaining events as required
+                bail!("unimplemented event {event:?}")
+            }
+        }
+    }
+
+    Ok(())
+}
```
Comment (Sec-ant) [inline xtask/rules_check/src/lib.rs:740]: I think we can remove most of the markdown content generation part in this function:

```suggestion
fn parse_documentation(
    group: &'static str,
    rule: &'static str,
    docs: &'static str,
    content: &mut Vec<u8>,
    has_fix_kind: bool,
) -> anyhow::Result<()> {
    let parser = Parser::new(docs);

    // Tracks the content of the current code block if it's using a
    // language supported for analysis
    let mut language = None;

    for event in parser {
        match event {
            // CodeBlock-specific handling
            Event::Start(Tag::CodeBlock(CodeBlockKind::Fenced(meta))) => {
                // Track the content of code blocks to pass them through the analyzer
                let test = CodeBlockTest::from_str(meta.as_ref())?;
                language = Some((test, String::new()));
            }
            Event::End(TagEnd::CodeBlock) => {
                if let Some((test, block)) = language.take() {
                    assert_lint(group, rule, &test, &block, has_fix_kind)?;
                }
            }
            Event::Text(text) => {
                if let Some((_, block)) = &mut language {
                    write!(block, "{text}")?;
                }
            }
            // We don't care other events
            _ => {}
        }
    }

    Ok(())
}
```

Comment (ematipico) [reply xtask/rules_check/src/lib.rs:740]: Did you test it locally?

Comment (Sec-ant) [reply xtask/rules_check/src/lib.rs:740]: > Did you test it locally?

No, but I can create a PR if this actually works.

Comment (ematipico) [reply xtask/rules_check/src/lib.rs:740]: No, it's OK, I can do a follow-up

**Pull request merged**